### ISSUE TITLE
Add minibatches to the data tool

### ISSUE THREAD

**Issue by perellonieto** [IRC-SPHERE/SklearnHyperStream#1]
It would be interesting to add the option of generating minibatches of data from the data tool. At this moment, each stream contains only one instance.
It does not seem difficult to implement.

**Comment by tdiethe** [comment]
There is already functionality for [sliding windows](https://github.com/IRC-SPHERE/HyperStream/blob/master/hyperstream/tools/sliding_window/2016-09-05_v0.0.1.py), so it shouldn't be too hard to instead have a sliding count. More complex batching (stratification/randomisation) would of course be trickier

**Comment by perellonieto** [comment]
I took a look at the sliding window. However, the way that is intuitive for me to use it would be:

1. Get any stream
2. Apply a sliding window to this stream and generate a new one
3. Use the later stream for the computations

The previous idea could be implemented in the following way:

```Python
    data_tool.execute(sources=[], sink=data_stream, interval=ti)

    mini_batch_tool = hs.tools.sliding_window(lower=-5, increment=5)
    mini_batch_stream = M.get_or_create_stream('mini_batch')
    mini_batch_tool.execute(sources=[data_stream], sink=mini_batch_stream,
                            interval=ti)
```

However, at this moment the sliding_window tool is not expecting any input stream.

The correct way to use it is depicted in the following example [sleep_detector.ipynb][1] but I don't find it that intuitive.

[1]: https://github.com/IRC-SPHERE/SPHERE-HyperStream/blob/d9bbde82ba6451400c6c6a3adc0408a894264aab/scripts/Sleep%20detector.ipynb

**Comment by perellonieto** [comment]
The commit 29c751e196a97796c167b302e98e30d2ae394184 implements the minibatch tool. Now the examples can be trained with mini-batches instead of stochastic gradient descent at single samples.

This decreases the training time several orders of magnitude depending on the dataset and minibatch size.

e.g. 

```Bash
time python example_classifier_keras.py --dataset digits --classifier mlp30ds40m --epochs 20 --seed 42 --batchsize 1 --learning-rate 1.0
```
End of the output:

```
Test scores per epoch
[ 0.4   0.69  0.79  0.83  0.85  0.86  0.88  0.89  0.9   0.91  0.91  0.92
  0.92  0.93  0.92  0.93  0.93  0.93  0.93  0.94]

real    0m33.454s
user    0m32.512s
sys     0m0.764s
```

Compared to:

```Bash
time python example_classifier_keras.py --dataset digits --classifier mlp30ds40m --epochs 20 --seed 42 --batchsize 100 --learning-rate 0.9
```
End of the output:

```Bash
Test scores per minibatch (cyclic)
[[ 0.21  0.25  0.22  0.13  0.18  0.21  0.17  0.26  0.3   0.28  0.25  0.29
   0.18  0.28  0.27  0.24  0.33  0.36  0.4   0.32  0.33  0.29  0.35  0.33
   0.31  0.4   0.4   0.43  0.34  0.35  0.37  0.39  0.39  0.43  0.43  0.49
   0.48  0.42  0.41  0.5   0.44  0.44  0.51  0.54  0.6   0.53  0.52  0.53
   0.57  0.49  0.47  0.59  0.6   0.62  0.65  0.52  0.58  0.64  0.5   0.52
   0.62  0.67  0.68  0.68  0.54  0.61  0.67  0.53  0.59  0.67  0.74  0.7
   0.73  0.58  0.62  0.67  0.59  0.62  0.69  0.75  0.75  0.71  0.64  0.65
   0.68  0.61  0.66  0.73  0.75  0.79  0.76  0.7   0.69  0.69  0.64  0.68
   0.74  0.74  0.82  0.79  0.69  0.73  0.71  0.62  0.74  0.77  0.76  0.84
   0.82  0.73  0.76  0.73  0.65  0.76  0.78  0.76  0.85  0.83  0.74  0.75
   0.76  0.67  0.8   0.8   0.76  0.87  0.85  0.76  0.78  0.78  0.67  0.84
   0.84  0.79  0.89  0.86  0.76  0.8   0.81  0.71  0.85  0.86  0.78  0.89
   0.87  0.77  0.78  0.78  0.71  0.85  0.88  0.79  0.89  0.88  0.8   0.81
   0.82  0.71  0.87  0.87  0.79  0.9   0.89  0.82  0.8   0.8   0.73  0.88
   0.88  0.8   0.9   0.86  0.8   0.79  0.8   0.75  0.91  0.88  0.83  0.92]]

real    0m7.329s
user    0m6.668s
sys     0m0.660s
```

Bear in mind that the learning rate needs to be adjusted depending on the mini-batch size.
